.github/workflows/yocto-build-deploy.yml

```diff
@@ -490,6 +490,25 @@ jobs:
           EOF
           cat "${AUTO_CONF_FILE}"
 
+      # https://docs.yoctoproject.org/4.0.10/ref-manual/classes.html?highlight=source_mirror#own-mirrors-bbclass
+      # https://github.com/openembedded/openembedded/blob/master/classes/own-mirrors.bbclass
+      # The own-mirrors class makes it easier to set up your own PREMIRRORS from which to first fetch source before
+      # attempting to fetch it from the upstream specified in SRC_URI within each recipe.
+      - name: Add S3 shared-downloads to PREMIRRORS
+        env:
+          SOURCE_MIRROR_URL: https://${{ vars.AWS_S3_BUCKET || vars.S3_BUCKET }}.s3.${{ vars.AWS_REGION || 'us-east-1' }}.amazonaws.com/shared-downloads/
+        run: |
+          mkdir -p "$(dirname "${AUTO_CONF_FILE}")"
+          cat <<EOF >> "${AUTO_CONF_FILE}"
+
+          INHERIT += "own-mirrors"
+          SOURCE_MIRROR_URL = "${SOURCE_MIRROR_URL}"
+
+          # BB_FETCH_PREMIRRORONLY = "1"
```

Comment:
nit: same here

Reply:
I used it for testing to make sure the premirrors were being used. I can remove it.